src/nnbench/types.py

```diff
@@ -20,13 +20,14 @@ def NoOp(**kwargs: Any) -> None:
 class BenchmarkRecord:
     context: Context
     benchmarks: list[dict[str, Any]]
-    _ctx_keys: tuple[str, ...] = field(init=False, repr=False)
 
-    def __post_init__(self):
-        super().__setattr__("_ctx_keys", tuple(self.context.keys()))
+    def contextkeys(self, sep: str = ".") -> tuple[str, ...]:
+        return tuple(self.context.keys(sep=sep))
```

Comment:
I just removed this in the Streamlit PR following your suggestion. But i prefer it this way. It's also necessary as we need to pass the `sep` for BigQuery, right? 

Let's merge this one before the Streamlit example and I'll make this change supercede the deletion and test for remaining issues with Prefect/Streamlit or merge conflicts. 

Anything else, I should know of w.r.t. to `contextkeys()`